### PR TITLE
update(events): World Event Dates/Times

### PR DESCRIPTION
**Pull request by CuriousMagpie**
- Updated world event dates and times to be consistent
- Added gem promos to repeating world events
- Added G1G1 to repeating world events

### CODE REVIEW

File: website/common/script/content/constants/events.js [hunk 1]
```diff
@@ -72,24 +72,39 @@ export const REPEATING_EVENTS = {
...
   harvestFeast: {
-    start: new Date('1970-11-22T08:00-05:00'),
-    end: new Date('1970-11-27T20:00-05:00'),
+    start: new Date('1970-11-22T04:00-05:00'),
+    end: new Date('1970-11-27T03:59-05:00'),
     season: 'thanksgiving',
     npcImageSuffix: '_thanksgiving',
     foodSeason: 'Pie',
   },
+  giveOneGetOne: {
```
Comment (phillipthelen) [inline website/common/script/content/constants/events.js]: The event is `giftOneGetOne`

File: website/common/script/content/constants/events.js [hunk 2]
```diff
@@ -72,24 +72,39 @@ export const REPEATING_EVENTS = {
...
+    start: new Date('1970-12-19T04:00-05:00'),
+    end: new Date('1971-01-06T03:59-05:00'),
```
Comment (phillipthelen) [inline website/common/script/content/constants/events.js]: the year is always 1970.

File: website/common/script/content/constants/events.js [hunk 3]
```diff
@@ -11,8 +11,8 @@ const gemsPromo = {
 
 export const REPEATING_EVENTS = {
   nye: {
-    start: new Date('1970-12-28T08:00-05:00'),
-    end: new Date('1970-01-04T23:59-05:00'),
+    start: new Date('1970-12-28T04:00-05:00'),
+    end: new Date('1971-01-04T03:59-05:00'),
```
Comment (phillipthelen) [inline website/common/script/content/constants/events.js]: the year is always 1970.

File: website/common/script/content/constants/events.js [hunk 4]
```diff
@@ -72,24 +72,39 @@ export const REPEATING_EVENTS = {
...
     foodSeason: 'Cake',
   },
+  fallGemFest: {
+    start: new Date('1970-09-23T04:00-04:00'),
+    end: new Date('1970-09-27T03:59-04:00'),
+    gemsPromo,
+  },
+  spookyGemFest: {
+    start: new Date('1970-10-28T04:00-04:00'),
+    end: new Date('1970-10-27T03:59-04:00'),
```
Comment (phillipthelen) [inline website/common/script/content/constants/events.js]: End date is before start date. This causes the event to span the entire year. Needs to end November 1st

File: website/common/script/content/constants/events.js [hunk 5]
```diff
@@ -72,24 +72,39 @@ export const REPEATING_EVENTS = {
...
+    gemsPromo,
+  },
   habitoween: {
-    start: new Date('1970-10-30T08:00-05:00'),
-    end: new Date('1970-11-01T23:59-05:00'),
+    start: new Date('1970-10-30T04:00-05:00'),
+    end: new Date('1970-11-01T23:059-05:00'),
```
Comment (phillipthelen) [inline website/common/script/content/constants/events.js]: The extra 0 here messes up the date parsing causing the end date to run into October

File: website/common/script/content/constants/events.js [hunk 6]
```diff
@@ -25,15 +25,15 @@ export const REPEATING_EVENTS = {
     ],
   },
   birthday: {
-    start: new Date('1970-01-30T08:00-05:00'),
-    end: new Date('1970-02-08T23:59-05:00'),
+    start: new Date('1970-01-30T04:00-05:00'),
+    end: new Date('1970-02-08T03:59-05:00'),
```
Comment (phillipthelen) [inline website/common/script/content/constants/events.js]: end date is February 1st

File: website/common/script/content/constants/events.js [hunk 7]
```diff
@@ -46,13 +46,13 @@ export const REPEATING_EVENTS = {
     ],
   },
   piDay: {
-    start: new Date('1970-03-13T08:00-05:00'),
-    end: new Date('1970-03-15T23:59-05:00'),
+    start: new Date('1970-03-13T04:00-05:00'),
```
Comment (phillipthelen) [inline website/common/script/content/constants/events.js]: Start date is march 14th. Timezone offset is `-04:00`

File: website/common/script/content/constants/events.js [hunk 8]
```diff
@@ -46,13 +46,13 @@ export const REPEATING_EVENTS = {
...
+    end: new Date('1970-03-15T03:59-05:00'),
     foodSeason: 'Pie',
   },
   aprilFoolsResale: {
-    start: new Date('1970-04-07T08:00-05:00'),
-    end: new Date('1970-04-30T23:59-05:00'),
+    start: new Date('1970-04-07T04:00-05:00'),
```
Comment (phillipthelen) [inline website/common/script/content/constants/events.js]: Timezone offset is `-04:00`

File: website/common/script/content/constants/events.js [hunk 9]
```diff
@@ -46,13 +46,13 @@ export const REPEATING_EVENTS = {
...
+    end: new Date('1970-04-30T03:59-05:00'),
```
Comment (phillipthelen) [inline website/common/script/content/constants/events.js]: End date is May 1st. Timezone offset is `-04:00`

File: website/common/script/content/constants/events.js [hunk 10]
```diff
@@ -72,24 +72,39 @@ export const REPEATING_EVENTS = {
     ],
   },
   namingDay: {
-    start: new Date('1970-07-30T08:00-05:00'),
-    end: new Date('1970-08-01T23:59-05:00'),
+    start: new Date('1970-07-30T04:00-05:00'),
```
Comment (phillipthelen) [inline website/common/script/content/constants/events.js]: Start date is July 31st. Timezone offset is `-04:00`

File: website/common/script/content/constants/events.js [hunk 11]
```diff
@@ -72,24 +72,39 @@ export const REPEATING_EVENTS = {
...
+    end: new Date('1970-08-01T03:59-05:00'),
```
Comment (phillipthelen) [inline website/common/script/content/constants/events.js]: End date is August 2nd. Timezone offset is `-04:00`

File: website/common/script/content/constants/events.js [hunk 12]
```diff
@@ -72,24 +72,39 @@ export const REPEATING_EVENTS = {
...
     foodSeason: 'Cake',
   },
+  fallGemFest: {
+    start: new Date('1970-09-23T04:00-04:00'),
+    end: new Date('1970-09-27T03:59-04:00'),
```
Comment (phillipthelen) [inline website/common/script/content/constants/events.js]: End time is `23:59`.

File: website/common/script/content/constants/events.js (hunk 4 above)
Comment (phillipthelen) [reply website/common/script/content/constants/events.js]: End time is `23:59`.

File: website/common/script/content/constants/events.js [hunk 13]
```diff
@@ -72,24 +72,39 @@ export const REPEATING_EVENTS = {
...
+  },
+  spookyGemFest: {
+    start: new Date('1970-10-28T04:00-04:00'),
+    end: new Date('1970-10-27T03:59-04:00'),
+    gemsPromo,
+  },
   habitoween: {
-    start: new Date('1970-10-30T08:00-05:00'),
-    end: new Date('1970-11-01T23:59-05:00'),
+    start: new Date('1970-10-30T04:00-05:00'),
```
Comment (phillipthelen) [inline website/common/script/content/constants/events.js]: Timezone offset is `-04:00`

File: website/common/script/content/constants/events.js [hunk 14]
```diff
@@ -72,24 +72,39 @@ export const REPEATING_EVENTS = {
...
+    end: new Date('1970-11-01T23:059-05:00'),
     foodSeason: 'Candy',
     season: 'habitoween',
     npcImageSuffix: '_halloween',
   },
   harvestFeast: {
-    start: new Date('1970-11-22T08:00-05:00'),
-    end: new Date('1970-11-27T20:00-05:00'),
+    start: new Date('1970-11-22T04:00-05:00'),
```
Comment (phillipthelen) [inline website/common/script/content/constants/events.js]: Start date is November 20th.

File: website/common/script/content/constants/events.js [hunk 15]
```diff
@@ -72,24 +72,39 @@ export const REPEATING_EVENTS = {
...
+    end: new Date('1970-11-27T03:59-05:00'),
```
Comment (phillipthelen) [inline website/common/script/content/constants/events.js]: End date is December 1st.

File: website/common/script/content/constants/events.js (hunk 2 above)
Comment (phillipthelen) [reply website/common/script/content/constants/events.js]: End time is `23:59`.

File: website/common/script/content/constants/events.js [hunk 16]
```diff
@@ -46,13 +46,13 @@ export const REPEATING_EVENTS = {
     ],
   },
   piDay: {
-    start: new Date('1970-03-13T08:00-05:00'),
-    end: new Date('1970-03-15T23:59-05:00'),
+    start: new Date('1970-03-13T04:00-05:00'),
+    end: new Date('1970-03-15T03:59-05:00'),
```
Comment (phillipthelen) [inline website/common/script/content/constants/events.js]: End date is march 16th. Timezone offset is `-04:00`

File: website/common/script/content/constants/events.js [hunk 17]
```diff
@@ -25,15 +25,15 @@ export const REPEATING_EVENTS = {
...
+    start: new Date('1970-01-30T04:00-05:00'),
+    end: new Date('1970-02-08T03:59-05:00'),
     season: 'birthday',
     npcImageSuffix: '_birthday',
     foodSeason: 'Cake',
   },
   valentines: {
-    start: new Date('1970-02-13T08:00-05:00'),
-    end: new Date('1970-02-17T23:59-05:00'),
+    start: new Date('1970-02-13T04:00-05:00'),
```
Comment (phillipthelen) [inline website/common/script/content/constants/events.js]: Start date is February 10th.

**Review by phillipthelen — CHANGES_REQUESTED** [review]
Additionally, in [this line](https://github.com/HabitRPG/habitica/blob/d3b63abdd33c2de769e44b277addb5314dfbd986/test/content/events.test.js#L13) in the events test, the faked date has to be changed from January 6th to p.e. January 8th.

File: website/common/script/content/constants/events.js (hunk 3 above)
Comment (CuriousMagpie) [reply website/common/script/content/constants/events.js]: I had some wrong logic in my head here

File: website/common/script/content/constants/events.js (hunk 7 above)
Comment (CuriousMagpie) [reply website/common/script/content/constants/events.js]: Daylight Savigs Time is the worst.

File: website/common/script/content/constants/events.js (hunk 1 above)
Comment (CuriousMagpie) [reply website/common/script/content/constants/events.js]: Which I get wrong approximately 50% of the tiime, regardless.

File: website/common/script/content/constants/events.js (hunk 2 above)
Comment (CuriousMagpie) [reply website/common/script/content/constants/events.js]: Again with wrong year logic in my head

File: website/common/script/content/constants/events.js [hunk 18]
```diff
@@ -46,13 +46,13 @@ export const REPEATING_EVENTS = {
     ],
   },
   piDay: {
-    start: new Date('1970-03-13T08:00-05:00'),
-    end: new Date('1970-03-15T23:59-05:00'),
+    start: new Date('1970-03-13T04:00-04:00'),
```
Comment (phillipthelen) [inline website/common/script/content/constants/events.js]: march 14th.

File: website/common/script/content/constants/events.js [hunk 19]
```diff
@@ -46,13 +46,13 @@ export const REPEATING_EVENTS = {
...
+    end: new Date('1970-03-16T03:59-05:00'),
```
Comment (phillipthelen) [inline website/common/script/content/constants/events.js]: UTC offset of 4 hours.

**Review by phillipthelen — CHANGES_REQUESTED** [review]

**Comment by phillipthelen** [comment]
The failing tests are due to the g1g1 event.

The ones that are failing bc it is expecting 1 event instead of 2 just needs the numbers changed to 2.
The one failing bc no event Is supposed to be returned needs a later date for checking.

**Review by phillipthelen — APPROVED** [review]

**Pull request merged**